src/v/cluster/topic_table.h

```diff
@@ -262,6 +262,42 @@ class topic_table {
     using delta_cb_t = ss::noncopyable_function<void(delta_range_t)>;
     using lw_cb_t = ss::noncopyable_function<void()>;
 
+    /// A helper struct that has various replica-related metadata all in one
+    /// place, so that the API user doesn't have to query several maps manually.
+    ///
+    /// Note that it contains references to current topic_table state and
+    /// therefore cannot be used across scheduling points.
+    struct partition_replicas_info {
```

Comment:
nit: i am wondering if we can use a `_view` suffix in the name of this type, this would indicate volatility

Reply:
I like it!